lib/ConstExtract/ConstExtract.cpp

```diff
@@ -119,28 +120,68 @@ parseProtocolListFromFile(StringRef protocolListFilePath,
   return true;
 }
 
+static std::string extractLiteralOutput(Expr *expr) {
+  std::string LiteralOutput;
+  llvm::raw_string_ostream OutputStream(LiteralOutput);
+  expr->printConstExprValue(&OutputStream, nullptr);
+
+  return LiteralOutput;
+}
+
 static std::shared_ptr<CompileTimeValue>
 extractPropertyInitializationValue(VarDecl *propertyDecl) {
   auto binding = propertyDecl->getParentPatternBinding();
   if (binding) {
     auto originalInit = binding->getOriginalInit(0);
     if (originalInit) {
-      std::string LiteralOutput;
-      llvm::raw_string_ostream OutputStream(LiteralOutput);
-      originalInit->printConstExprValue(&OutputStream, nullptr);
-      if (!LiteralOutput.empty())
+      auto LiteralOutput = extractLiteralOutput(originalInit);
+      if (!LiteralOutput.empty()) {
         return std::make_shared<RawLiteralValue>(LiteralOutput);
+      }
+
+      if (auto callExpr = dyn_cast<CallExpr>(originalInit)) {
+        std::vector<FunctionParameter> parameters;
+
+        const auto args = callExpr->getArgs();
+        for (auto arg : *args) {
+          auto label = arg.getLabel().str().str();
+          auto expr = arg.getExpr();
+
+          switch (expr->getKind()) {
+          case ExprKind::DefaultArgument: {
+            auto defaultArgument = dyn_cast<DefaultArgumentExpr>(expr);
+            auto *paramDecl = defaultArgument->getParamDecl();
+
+            if (paramDecl->hasDefaultExpr()) {
+              LiteralOutput =
+                  extractLiteralOutput(paramDecl->getTypeCheckedDefaultExpr());
+            }
+
+            break;
+          }
+          default:
+            LiteralOutput = extractLiteralOutput(expr);
```

Comment:
We may eventually need to introduce recursion — instead of just calling `extractLiteralOutput()` — to support kinds like Dictionary and even other statically-extractable init/builder calls.

Reply:
Yes, I completely agree.